saw-core-coq/src/SAWCoreCoq/Term.hs

```diff
@@ -579,6 +599,35 @@ translateBinder n ty@(asPiList -> (args, pi_body)) f =
                 helper n' rest (g . ((nh,nhty) :))
          helper n' ((False,_):rest) g = helper n' rest g
 
+-- | Given an 'ExtCns' with its type (as a 'Term'), translate the 'ExtCns'
+-- to a Coq identifier, translate the type to a Coq term, and generate zero or
+-- more additional 'Ident's and 'Type's representing additonal implicit
+-- typeclass arguments, added if the given type is @isort@, etc. Pass all of
+-- this information to the supplied computation, in which the SAW core variable
+-- is bound to its Coq identifier.
+translateBinderEC ::
+  TermTranslationMonad m => ExtCns Term -> (BindTrans -> m a) -> m a
+translateBinderEC ec f =
+  do ty' <- translateTerm ty
+     let mb_sort = asSortWithFlags pi_body
+         flagValues = sortFlagsToList $ maybe noFlags snd mb_sort
+         flagLocalNames = [("Inh", "SAWCoreScaffolding.Inhabited"),
+                           ("QT", "QuantType")]
+     withSAWVarEC nm $ \n' ->
+       helper n' (zip flagValues flagLocalNames) (\imps ->
+                                                   f $ BindTrans n' ty' imps)
+       where
+         ty = ecType ec
+         (args, pi_body) = asPiList ty
+         nm = Name (ecVarIndex ec) (ecName ec)
+         n = toShortName (ecName ec)
+         helper _ [] g = g []
+         helper n' ((True,(prefix,tc)):rest) g =
+           do nhty <- translateImplicitHyp (Coq.Var tc) args (Coq.Var n')
+              withFreshIdent (prefix <> "_" <> n) $ \nh ->
+                helper n' rest (g . ((nh,nhty) :))
+         helper n' ((False,_):rest) g = helper n' rest g
```

Comment:
`helper` would benefit from having a type signature, especially given the use continuation-passing style.

Reply:
Fixed in b4e344f69f8fb86660bdc1db30a317efef461d6e.